packages/components/src/internal/components/domainproperties/samples/UniqueIdBanner.spec.tsx

```diff
@@ -0,0 +1,219 @@
+                rowId: 1,
+            }),
+            domainDesign: DomainDesign.create({
+                fields: [intField, stringField],
+            })
+        } as DomainDetails);
+        const wrapper = mount(
+            <UniqueIdBanner
+                model={sampleTypeModel}
+                isFieldsPanel={false}
+                onAddField={jest.fn}
+            />
+        )
+        expect(wrapper.find(Alert)).toHaveLength(1);
+        expect(wrapper.find(Button)).toHaveLength(1);
+        expect(wrapper.text()).toContain(ADD_NEW_UNIQUE_ID_MSG);
+        wrapper.unmount();
+    });
+
+    test("not new, in fields panel, no uniqueID fields", () => {
+        const sampleTypeModel = SampleTypeModel.create({
+            options: fromJS({
+                rowId: 1,
+            }),
+            domainDesign: DomainDesign.create({
+                fields: [intField, stringField],
+            })
+        } as DomainDetails);
+        const wrapper = mount(
+            <UniqueIdBanner
+                model={sampleTypeModel}
+                isFieldsPanel={true}
+                onAddField={jest.fn}
+            />
+        )
+        expect(wrapper.find(Alert)).toHaveLength(1);
+        expect(wrapper.find(Button)).toHaveLength(1);
+        expect(wrapper.text()).toContain(ADD_NEW_UNIQUE_ID_MSG);
+        wrapper.unmount();
+    });
+
+    test("not new, in properties panel, with uniqueID fields", () => {
+        const sampleTypeModel = SampleTypeModel.create({
+            options: fromJS({
+                rowId: 1,
+            }),
+            domainDesign: DomainDesign.create({
+                fields: [intField, stringField, uniqueIdField, uniqueIdField2],
+            })
+        } as DomainDetails);
+        const wrapper = mount(
+            <UniqueIdBanner
+                model={sampleTypeModel}
+                isFieldsPanel={false}
+                onAddField={jest.fn}
+            />
+        )
+        expect(wrapper.find(Alert)).toHaveLength(0);
+        expect(wrapper.find(Button)).toHaveLength(0);
+        expect(wrapper.text()).toContain("2 Unique ID fields are defined: " + uniqueIdField.name + ", "  + uniqueIdField2.name);
+
+        wrapper.unmount();
+    });
+
+    test("not new, in fields panel, with uniqueID fields", () => {
+        const sampleTypeModel = SampleTypeModel.create({
+            options: fromJS({
+                rowId: 1,
+            }),
+            domainDesign: DomainDesign.create({
+                fields: [intField, stringField, uniqueIdField, uniqueIdField2],
+            })
+        } as DomainDetails);
+        const wrapper = mount(
+            <UniqueIdBanner
+                model={sampleTypeModel}
+                isFieldsPanel={true}
+                onAddField={jest.fn}
+            />
+        )
+        expect(wrapper.find(Alert)).toHaveLength(0);
+        expect(wrapper.find(Button)).toHaveLength(0);
+        expect(wrapper.text()).toHaveLength(0);
```

Comment:
This case where all of the test cases are checking the same things with different expected values is where I have recently been adding in the `validate()` helper function.